src/widgets/JSONForm/JSONForm.js

```diff
@@ -113,51 +127,97 @@ class FocusController {
   };
 }
 
-export const JSONForm = React.forwardRef(({ theme = defaultTheme, children, options }, ref) => {
-  const formRef = useRef(null);
-
-  const Form = useMemo(() => withTheme(theme), []);
-
-  // Attach to the ref passed a method `submit` which will allow a caller
-  // to programmatically call submit
-  useImperativeHandle(ref, () => ({
-    submit: e => {
-      formRef.current?.onSubmit(e);
-    },
-  }));
-
-  return (
-    <JSONFormContext.Provider value={options}>
-      <Form
-        uiSchema={testTongaUiSchema}
-        onError={() => {
-          // placeholder to prevent console.errors when validation fails.
-        }}
-        onSubmit={form => {
-          Alert.alert('Form Data', JSON.stringify(form.formData), null, '\n');
-        }}
-        ref={formRef}
-        schema={testTongaSurvey}
-      >
-        {children ?? (
-          <PageButton
-            onPress={e => {
-              formRef.current?.onSubmit(e);
-            }}
-          />
-        )}
-      </Form>
-    </JSONFormContext.Provider>
-  );
-});
+export const JSONForm = React.forwardRef(
+  ({ theme, children, options, jsonSchema, uiSchema }, ref) => {
+    const validator = useMemo(() => ajv.compile(jsonSchema), [jsonSchema]);
+    const Form = useMemo(() => withTheme(theme), []);
+
+    // Attach to the ref passed a method `submit` which will allow a caller
+    // to programmatically call submit
+    const formRef = useRef(null);
+    useImperativeHandle(
+      ref,
+      () => ({
+        submit: e => {
+          formRef.current?.onSubmit(e);
+        },
+      }),
+      []
+    );
+
+    return (
+      <JSONFormContext.Provider value={options}>
+        <Form
+          validate={(formData, errorHandlers) => {
```

Comment:
This is all totally strange stuff to be able to add `ajv-errors` in .. :( Hopefully the comments help and not hinder